sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -61,55 +142,75 @@ impl BlobClient {
             .per_call_policies
             .push(storage_headers_policy);
 
-        let client = GeneratedBlobClient::new(
-            endpoint,
-            credential.clone(),
-            container_name.clone(),
-            blob_name.clone(),
-            Some(options),
-        )?;
-        Ok(Self {
-            endpoint: endpoint.parse()?,
-            client,
-        })
-    }
-
-    /// Returns a new instance of AppendBlobClient.
-    ///
-    /// # Arguments
-    ///
-    pub fn append_blob_client(&self) -> AppendBlobClient {
-        AppendBlobClient {
-            endpoint: self.client.endpoint.clone(),
-            client: self.client.get_append_blob_client(),
+        let mut url = Url::parse(endpoint)?;
+        if !url.scheme().starts_with("http") {
+            return Err(azure_core::Error::message(
+                azure_core::error::ErrorKind::Other,
+                format!("{url} must use http(s)"),
+            ));
         }
-    }
 
-    /// Returns a new instance of BlockBlobClient.
-    ///
-    /// # Arguments
-    ///
-    pub fn block_blob_client(&self) -> BlockBlobClient {
-        BlockBlobClient {
-            endpoint: self.client.endpoint.clone(),
-            client: self.client.get_block_blob_client(),
-        }
+        // In regular ctor, since struct now holds blob_url, we have to build the blob_url given endpoint + container_name + blob_name
+        url.path_segments_mut()
+            .expect("Cannot be base")
+            .extend([&container_name, &blob_name]);
+
+        let client =
+            GeneratedBlobClient::from_url_with_credential(url.as_str(), credential, Some(options))?;
+        Ok(Self { client })
     }
 
-    /// Returns a new instance of PageBlobClient.
-    ///
-    /// # Arguments
-    ///
-    pub fn page_blob_client(&self) -> PageBlobClient {
-        PageBlobClient {
-            endpoint: self.client.endpoint.clone(),
-            client: self.client.get_page_blob_client(),
-        }
+    pub fn from_blob_url(blob_url: &str, options: Option<BlobClientOptions>) -> Result<Self> {
+        let mut options = options.unwrap_or_default();
+
+        let storage_headers_policy = Arc::new(StorageHeadersPolicy);
+        options
+            .client_options
+            .per_call_policies
+            .push(storage_headers_policy);
+
+        let url = Url::parse(blob_url)?;
+        let client = GeneratedBlobClient::from_url_with_no_credential(url.as_str(), Some(options))?;
+
+        Ok(Self { client })
```

Comment:
Example of exposed `from_blob_url`. We could then add credential support (either completely discrete API, or Option<Cred> single API matching)